bin/deployer_keycloak

```diff
@@ -0,0 +1,249 @@
+
+
+# Publish message to ams upstream topic. Get as arguments
+# - the messages to be published
+# - the ams agent to handle the operation
+def publish_ams(pub_messages, ams_agent):
+    if len(pub_messages) > 0:
+        log.info("Publish messaged to ams")
+        log.debug("Messages published to ams: " + str(pub_messages))
+        ams_agent.publish(pub_messages)
+
+
+# Calls Keycloak depending on the deployment type provided.
+# Operations handled:
+# - create
+# - delete
+# - edit
+def call_keycloak(registry_message, keycloak_agent):
+    deployment_type = registry_message.pop("deployment_type")
+    keycloak_msg = format_keycloak_msg(registry_message, deployment_type)
+    log.debug("Formated message for Keycloak: " + str(keycloak_msg))
+    response = {}
+    external_id = -1
+    client_id = ""
+    if deployment_type == "create":
+        log.info("Create new client")
+        response = keycloak_agent.createClient(keycloak_msg)
+        if response["status"] == 200:
+            external_id = response["response"]["id"]
+            client_id = response["response"]["clientId"]
+    elif deployment_type == "delete":
+        external_id = registry_message["external_id"]
+        log.info("Delete client with id: " + str(external_id))
+        response = keycloak_agent.deleteClientById(registry_message["external_id"])
+    elif deployment_type == "edit":
+        external_id = registry_message["external_id"]
+        log.info("Update client with id: " + str(external_id))
+        response = keycloak_agent.updateClientById(
+            registry_message["external_id"], keycloak_msg
+        )
+        if response["status"] == 200:
+            client_id = response["response"]["clientId"]
+    return response, external_id, client_id
+
+
+if __name__ == "__main__":
+    # Get config path from arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-c", required=True, type=str, help="Configuration file location path"
+    )
+    args = parser.parse_args()
+    path = args.c
+    with open(path) as json_data_file:
+        config = json.load(json_data_file)
+
+    # Get log_conf from project arguments else use the global setting
+    if "log_conf" in config["keycloak"]:
+        get_log_conf(config["keycloak"]["log_conf"])
+    else:
+        get_log_conf(config["log_conf"])
+
+    log.info("Init ams agent")
+    ams = PullPublish(config["keycloak"]["ams"])
+
+    # Get messages
+    while True:
+        log.info("Pull messages from ams")
+        messages, ids = ams.pull(1)
+        log.info("Received " + str(len(messages)) + " messages from ams")
+        if len(messages) > 0:
+            log.info("Get access token from " + config["keycloak"]["issuer"])
+            access_token = clientCredentialsGrant(
+                config["keycloak"]["issuer"],
+                config["keycloak"]["client_id"],
+                config["keycloak"]["client_secret"],
+            )
+            ams.ack(ids)
```

Comment:
I would suggest to acknowledge the message after the update function to eliminate the message loss in case an update fail